src/condor_tools/htcondor_cli/credential.py

```diff
@@ -10,6 +10,129 @@
 from htcondor_cli.verb import Verb
 
 
+def compactify_job_list(jobs):
+    if jobs is None:
+        return ""
+
+    list = ""
+    for clusterID in sorted(jobs.keys()):
+        ordered = sorted(jobs[clusterID])
+
+        # This whole thing needs some help.
+        if len(ordered) == 1:
+            list += f"{clusterID}.{ordered[0]}; "
+        elif len(ordered) == 2:
+            list += f"{clusterID}.{ordered[0]},{ordered[1]}; "
+        else:
+            list += f"{clusterID}."
+
+            last = None
+            first = None
+            index = 0
+
+            while index < len(ordered):
+                if index + 1 == len(ordered):
+                    last = ordered[index]
+                    next = None
+                else:
+                    next = ordered[index + 1]
+
+                if ordered[index] + 1 == next:
+                    if first is None:
+                        first = ordered[index]
+                    last = ordered[index + 1]
+                else:
+                    if first is None:
+                        list += f"{ordered[index]},"
+                    else:
+                        list += f"[{first}-{last}],"
+                        first = None
+                        last = None
+                index += 1
+
+            list = list[:-1]
+            list += "; "
+
+    return list[:-2]
+
+
+class ListAll(Verb):
+    def __init__(self, logger, **options):
+        # Grovel around in SEC_CREDENTIAL_DIRECTORY_OAUTH.
+        credentials_dir_path = htcondor.param.get("SEC_CREDENTIAL_DIRECTORY_OAUTH")
+        if credentials_dir_path is None:
+            return
+        credentials_dir = Path(credentials_dir_path)
+
+
+        credentials_by_user = defaultdict(dict)
+        try:
+            for entry in credentials_dir.iterdir():
+                if not entry.is_dir():
+                    continue
+
+                for file in entry.iterdir():
+                    if file.suffix != ".top":
+                        continue
+                    credentials_by_user[entry.name][file.stem] = {
+                        'refresh':  int(file.stat().st_mtime),
+                        'jobs':     defaultdict(list),
+                    }
+        except PermissionError:
+            print("You don't have permission to read the credentials directory.")
```

Comment:
Could/should this message include the owner of the creddir?

Reply:
The owner of the creddir has been `condor.condor` on every machine I've checked so far (with the group sticky bit set, I guess?), but I'm pretty sure we don't want to tell people to su(do) to the condor user, so I'd rather not get into that.  I'd be fine with adding something like, "Try again as the root or admnistrator account".